src/test/java/com/exasol/adapter/dialects/exasol/AbstractExasolSqlDialectIT.java

```diff
@@ -0,0 +1,558 @@
+        assertCast("VARCHAR(30)", "DATE", "2020-10-17", java.sql.Date.valueOf("2020-10-17"));
+    }
+
+    @Test
+    void testCastVarcharasDecimal() {
+        assertCast("VARCHAR(6)", "DECIMAL(5,1)", "1234.5", BigDecimal.valueOf(1234.5));
+    }
+
+    @Test
+    void testCastVarcharAsDouble() {
+        assertCast("VARCHAR(6)", "DOUBLE PRECISION", "1234.5", 1234.5);
+    }
+
+    @Test
+    void testCastVarcharAsGeometry() {
+        assertCast("VARCHAR(20)", "GEOMETRY(5)", "VARCHAR", "POINT(2 5)", "POINT (2 5)");
+    }
+
+    @Test
+    void testCastVarcharAsTimestamp() {
+        final String timestampAsString = "2016-06-01 13:17:02.081";
+        assertCast("VARCHAR(30)", "TIMESTAMP", timestampAsString, Timestamp.valueOf(timestampAsString));
+    }
+
+    @Test
+    void testCastVarcharAsTimestampWithLocalTimezone() {
+        final String timestamp = "2017-11-03 14:18:02.081";
+        assertCast("VARCHAR(30)", "TIMESTAMP WITH LOCAL TIME ZONE", "TIMESTAMP", timestamp,
+                Timestamp.valueOf(timestamp));
+    }
+
+    @Test
+    void castIntegerAsVarchar() {
+        assertCast("INTEGER", "VARCHAR(5)", 12345, "12345");
+    }
+
+    @Test
+    void testCaseEqual() {
+        final Table table = createSingleColumnTable("INTEGER").insert(1).insert(2).insert(3);
+        this.virtualSchema = createVirtualSchema(this.sourceSchema);
+        assertVsQuery("SELECT CASE C1 WHEN 1 THEN 'YES' WHEN 2 THEN 'PERHAPS' ELSE 'NO' END FROM " //
+                + getVirtualTableName(this.virtualSchema, table), //
+                table().row("YES").row("PERHAPS").row("NO").matches());
+    }
+
+    @Test
+    void testCaseGreaterThan() {
+        final Table table = createSingleColumnTable("INTEGER").insert(1).insert(2).insert(3);
+        this.virtualSchema = createVirtualSchema(this.sourceSchema);
+        assertVsQuery("SELECT CASE WHEN C1 > 1 THEN 'YES' ELSE 'NO' END FROM " //
+                + getVirtualTableName(this.virtualSchema, table), //
+                table().row("NO").row("YES").row("YES").matches());
+    }
+
+    @Test
+    void testLeftJoin() {
+        createVirtualSchemaWithTablesForJoinTest();
+        assertVsQuery("SELECT * FROM TL LEFT JOIN TR ON TL.C1 = TR.C1 ORDER BY TL.C1", //
+                table() //
+                        .row("K1", "L1", "K1", "R1") //
+                        .row("K3", "L3", null, null) //
+                        .row(null, "L2", null, null) //
+                        .matches());
+    }
+
+    @Test
+    void testLeftJoinWithProjection() {
+        createVirtualSchemaWithTablesForJoinTest();
+        assertVsQuery("SELECT TL.C1, TL.C2, TR.C2 FROM TL LEFT JOIN TR ON TL.C1 = TR.C1 ORDER BY TL.C1", //
+                table() //
+                        .row("K1", "L1", "R1") //
+                        .row("K3", "L3", null) //
+                        .row(null, "L2", null) //
+                        .matches());
+    }
+
+    private void createVirtualSchemaWithTablesForJoinTest() {
```

Comment:
maybe we can move this method to line 468, right after the first place it's used